impl/src/main/java/com/sun/faces/util/Util.java

```diff
@@ -549,6 +559,31 @@ public static boolean isEmpty(String string) {
         return string == null || string.isEmpty();
     }
 
+    /**
+     * Returns true if the given string is not null and not empty.
+     *
+     * @param string The string to be checked on emptiness.
+     * @return True if the given string is not null and not empty.
+     */
+    public static boolean isNotEmpty(String string) { return !isEmpty(string); }
+
+    /**
+     * Returns true if the given string is null or is blank.
+     *
+     * @param string The string to be checked.
+     * @return True if the given string is null or is blank.
+     */
+    public static boolean isBlank(String string) {
+        return string == null || string.isBlank();
+    }
+
+    /**
+     * @return null if the passed String is null or blank, s otherwise
+     */
+    public static String nullIfBlank(String s) {
+        return isBlank(s) ? null : s;
+    }
+
```

Comment:
This is not an optimization, this is an addition which is nowhere used.

Reply:
ok